packages/ui/tests/Customization/ExtensionDetail.spec.tsx

```diff
@@ -79,22 +78,27 @@ export default describe('ExtensionDetail', () => {
 
     // usage section title
     expect(queryAllByText(usageLabel)).toHaveLength(1);
-
-    // test snapshot
-    const snapshot = renderer.create(comp).toJSON();
-    expect(snapshot).toMatchSnapshot();
   });
 
-  it('Should open delete confirmation modal', () => {
+  it('Should open delete confirmation modal', async () => {
+    // need to set extensionUses to zero so that the delete button is enabled
     const comp = <ExtensionDetail {...props} extensionUses={0} />;
     const { baseElement, getByText } = render(comp);
     const deleteButton = getByText(deleteLabel);
     expect(deleteButton).not.toHaveAttribute('disabled'); // delete should be enabled
-    fireEvent.click(deleteButton); // open delete confirmation dialog
-    expect(baseElement.classList).toContain('modal-open');
 
-    // test snapshot
-    const snapshot = renderer.create(comp).toJSON();
-    expect(snapshot).toMatchSnapshot();
+    // click the delete button so that the delete confirmation dialog opens
+    fireEvent.click(deleteButton);
+
+    // wait for the delete dialog to show by looking for the delete button
+    const elements = await waitForElement(() => {
+      // find the delete button
+      return baseElement.getElementsByClassName('btn btn-danger');
```

Comment:
You are testing the implementation this way. Why not querying for the `deleteLabel` text?

Reply:
Tried that but ran into the problem that there are 2 components in the DOM that have that text (the delete button in the popup confirmation dialog, and the delete button on the details component).

Reply:
Ah, gotcha. And since the dialog is just a wrapper over the one exposed by PF we can't add a `data-testid` to its button either. 

Maybe wrap the overlay in some div with a data-testid, query for that and then use getByText on that element? Something like this (warning: I haven't tested it lol):

```
import {getByText, ... } from 'react-testing-library';

...

const {getByTestId, ...} = render(...);
const dialog = getByTestId('dialog');
expect(getByText(dialog, deleteLabel)).toBeInTheDocument();
```

Reply:
Isn't that kind of knowing the implementation? Why would this way be better than what I originally had? (only saying this bc I'm having trouble getting your idea to work:)

Reply:
It's a fair question. It's not a problem knowing about the implementation, actually, you have to know how the thing works to test it, right? It's testing the implementation that's the problem. The way you have will break if another "danger" button gets added to the dialog, where with the suggested change the test will continue to pass.

I understand that my example is an unlikely scenario, but it's important to remember that a component should be tested from the user standpoint. A CSS class is invisible to a user, so it's better not to rely on it unless strictly necessary.

> On top of the queries provided by the testing library, you can using the regular querySelector DOM API to query elements. Note that using this as an escape hatch to query by class or id is a bad practice because users can't see or identify these attributes. Use a testid if you have to.

https://testing-library.com/docs/guide-which-query#manual-queries

Reply:
Modified test as we discussed @riccardo-forina.